gulpfile.js

```diff
@@ -94,29 +95,40 @@ gulp.task('imgmin', function () {
       .pipe(gulp.dest('public/img'));
 });
 
-// webpack build:
-gulp.task('webpack', function() {
-  webpack({
-    entry: './js/main.js',
-    output: {
-      path: __dirname + '/public/js',
-      filename: 'j.js'
-    },
-    plugins: [
-      new webpack.optimize.UglifyJsPlugin({
+// wrapper for JS build, supporting both
+// minified and unminified versions
+function jsBuild(isProduction) {
+  return gulp.src('js/main.js')
+    .pipe(webpackStream({
+      entry : {
+        j : './js/main.js'
+      },
+      output: {
+        filename: '[name].js'
```

Comment:
This is where gulp (or webpack-stream) is a little weird. `[name]` is a placeholder for the property name in the `entry` object above; so, the output goes to j.js for ./js/main.js. If you had additional property names in the `entry` object, that would result in additional output files. I think you could do this more simply -- if you only ever needed to generate one JS file -- but I did it this way because we'd talked about the ability to have different files for different pages.